index.d.ts

```diff
@@ -17,7 +23,7 @@ declare function createError (
   message: string,
   statusCode?: number,
   Base?: Error
-): FastifyError
+): FastifyErrorConstructor
 
 
-export default createError
+export default createError
```

Comment:
We need to add the "infamous triplet" here because the context used in the JS code is the CommonJS as we can see here https://github.com/fastify/fastify-error/blob/master/index.js#L43
When I say "infamous triplet" i am reffering to https://github.com/fastify/fastify/blob/master/fastify.js#L643, https://github.com/fastify/fastify/blob/master/fastify.js#L643 and https://github.com/fastify/fastify/blob/master/fastify.d.ts#L136

Using `export default` in the TS types and "module.exports =` in JS makes this package broken in several TS scenarios: for example when used with module interop false since it will not emit the importStar helper and will break runtime.

Reply:
not super certain what is infamous triplet but added something along the lines